corehq/apps/accounting/models.py

```diff
@@ -3958,20 +3964,39 @@ def clean(self):
             raise ValidationError(_("You can't specify both an invoice and a line item."))
 
 
-class DomainUserHistory(models.Model):
+class DomainUserHistoryBase(models.Model):
     """
-    A record of the number of users in a domain at the record_date.
-    Created by task calculate_users_in_all_domains on the first of every month.
-    Used to bill clients for the appropriate number of users
+    Base record of the number of mobile workers in a domain for a given date.
     """
     domain = models.CharField(max_length=256)
     record_date = models.DateField()
     num_users = models.IntegerField(default=0)
 
     class Meta:
+        abstract = True
         unique_together = ('domain', 'record_date')
 
 
+class DomainUserHistory(DomainUserHistoryBase):
+    """
+    The total number of mobile workers in a domain at the record_date.
+    Created by task calculate_users_in_all_domains on the first of every month.
+    Used to bill clients for the appropriate number of mobile workers.
+    """
+    pass
+
+
+class DomainSubmittingMobileWorkerHistory(DomainUserHistoryBase):
+    """
+    The number of form-submitting mobile workers in a domain for the month
+    preceeding the record date.
```

Comment:
Can you be explicit in the comment about what a month means in this context? 30 days? 31 days? However many days the current month is long? The full previous calendar month (i.e. not up through to day? The current calendar month up through today?

Reply:
> The full previous calendar month (i.e. not up through to day?

The intent is for it to be this one, using the `calculate` form-submitting mobile workers task that will be run on 1st of every month. There's nothing in the model limiting it to that though, and the way the task will likely be written (mirroring other similar invoicing functions), running the task on say, August 9th would use data from dates July 9th to August 8th, inclusive. 

Is it worth being that explicit in the comment, or is it reasonable just to call out the intended, practical use here?

Reply:
I think it's worth being explicit in the comment

Reply:
https://github.com/dimagi/commcare-hq/pull/34847/commits/1233633b8e6eacb7ae0656c2cd9de055205651df